tests/test_classdecorators.py

```diff
@@ -0,0 +1,411 @@
+import uuid
+
+import pytest
+import sqlalchemy as sa
+
+from dbos_transact.context import (
+    DBOSContextEnsure,
+    SetWorkflowUUID,
+    assert_current_dbos_context,
+)
+from dbos_transact.dbos import DBOS
+
+
+class DBOSTestClassInstNN:
+    @DBOS.required_roles(["user"])
+    def test_func(self, var: str) -> str:
+        return var
+
+
+class DBOSTestClassInst:
+    def __init__(self) -> None:
+        self.instance_name = "myname"
+
+    @DBOS.required_roles(["user"])
+    def test_func(self, var: str) -> str:
+        return var
+
+
+"""
+# This cannot work yet - no DBOS decorator available
```

Comment:
Clean up comments in this file?

Reply:
I want to be able to do those things though at some point.